### ISSUE TITLE
Static analyzer rules not included into PMD and Xcode reports

### ISSUE THREAD

**Issue by mgrebenets** [oclint/oclint#363]
Hi

I have created a sample code with intentional memory leak.
It is reported by Xcode's "Analyze" action, it is properly picked up by OCLint versions `0.10.2` and `0.10.3` and reported for the following report types: `text`, `xml`, `html`, `json`.
For `pmd` and `xcode` reports, however, the static analyzer error is not included into the report.
For me it's most important to get it in `pmd` report, so that I can upload it to SonarQube then.

Is it an intentional limitation or is it a bug?

**Comment by mgrebenets** [comment]
After playing around with it and trying to manually stick static analyzer violations into PMD report file, I noticed that sonar qube runner fails with errors like "Rule must be set" or "unknown rule" if I try to add `rule=` attribute with fake name.

However, I was able to include static analyzer errors using some valid rune name, for example:

```xml
<violation beginline="29" begincolumn="18" priority="5" rule="unnecessary else statement">Potential leak of memory pointed to by 'p'</violation>
```

The it shows up in SonarQube like so:

![image](https://cloud.githubusercontent.com/assets/350654/17351319/d707be38-5970-11e6-9f9a-c55a29f26c4b.png)

I guess Sonar plugin is checking the rule name agains this list: https://github.com/octo-technology/sonar-objective-c/blob/master/src/main/resources/org/sonar/plugins/oclint/profile-oclint.xml

Could there be a solution for this?

Could it be possible to add new OCLint rule with name "clang static analyzer" and add all static analyzer errors under this rule name to PMD report?
As a matter of fact, other report types like `xml`, `html` and so on already do exactly the same thing:

![image](https://cloud.githubusercontent.com/assets/350654/17351371/34311fe6-5971-11e6-8f4e-6e434daafa59.png)

So I guess if new rule would be added to OCLint rules list and then added to PMD reports, the plugins like Sonar plugin would follow and will be able to parse and display it properly.

**Comment by UnrealQuester** [comment]
Implemented in PR #367.

**Comment by ryuichis** [comment]
I would like to discuss whether clang static analyzer results should be included in the xcode-reporter. I can see both sides. On one side, we can invoke clang static analyzer directly inside Xcode w/o running OCLint. On the other side, we don't want to waste the time and resources by invoking two tools. Thoughts? @UnrealQuester @mgrebenets

**Comment by UnrealQuester** [comment]
Including the static analyzer bugs is just a couple of lines additional code and does not cost us anything  when `-enable-clang-static-analyzer` is not set. I could open up another pull request for that if you want me to.

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by ryuichi-assistant** [comment]
This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to open a new one if you still experience this problem.
